appmap/_implementation/generation.py

```diff
@@ -0,0 +1,117 @@
+"""Generate an AppMap"""
+import json
+from collections.abc import MutableMapping
+
+from . import metadata
+
+from .event import Event, serialize_event
+
+
+class ClassMapSet(MutableMapping):
```

Comment:
Doesn't seem to be doing anything specifically different from `dict`. Subclassing `dict` might give similar behavior.

Reply:
I guess this was based on a misunderstanding of when to subclass `dict` vs when to subclass `MutableMapping`. I'll switch to `dict.

Reply:
based on the current implementation it looks like we are just renaming `dict` to ClassMapSet. If you think in future, there could be more custom code in the getter setter or some other internal implementation can leave it as is.

Reply:
You're right about the intention here, which just amounts to renaming `dict`. However, when I switched to `class ClassMapSet(dict)`,  the tests failed. So, I think I'll leave this as-is, for now, and switch to subclassing `dict` in a future round of changes.